File: fabric/src/main/java/io/github/xiewuzhiying/vs_addition/fabric/VSAdditionModFabric.java
package io.github.xiewuzhiying.vs_addition.fabric;

import dan200.computercraft.api.peripheral.PeripheralLookup;
import io.github.xiewuzhiying.vs_addition.VSAdditionMod;
import io.github.xiewuzhiying.vs_addition.stuff.EntityFreshCaller;
import io.xiewuzhiying.vs_addition.fabric.compats.computercraft.FabricPeripheralLookup;
import net.fabricmc.api.ModInitializer;
import net.fabricmc.fabric.api.event.lifecycle.v1.ServerEntityEvents;
import org.valkyrienskies.mod.fabric.common.ValkyrienSkiesModFabric;

import static io.github.xiewuzhiying.vs_addition.VSAdditionMod.init;
import static io.github.xiewuzhiying.vs_addition.compats.computercraft.PeripheralCommon.registerGenericPeripheralCommon;

public class VSAdditionModFabric implements ModInitializer {
    @Override
    public void onInitialize() {
        // force VS2 to load before eureka
        new ValkyrienSkiesModFabric().onInitialize();

        init();

        if(VSAdditionMod.getCC_ACTIVE()) {
            registerGenericPeripheralCommon();
            PeripheralLookup.get().registerFallback((level, blockPos, blockState, blockEntity, direction) -> FabricPeripheralLookup.INSTANCE.peripheralProvider(level, blockPos));
        }

        ServerEntityEvents.ENTITY_LOAD.register((EntityFreshCaller.INSTANCE::freshEntityInShipyard));
    }
}

File: fabric/src/main/kotlin/io/github/xiewuzhiying/vs_addition/fabric/VSAdditionModFabric.kt
package io.github.xiewuzhiying.vs_addition.fabric

import io.github.xiewuzhiying.vs_addition.VSAdditionMod.CC_ACTIVE
import io.github.xiewuzhiying.vs_addition.VSAdditionMod.init
import io.github.xiewuzhiying.vs_addition.VSAdditionMod.initClient
import io.github.xiewuzhiying.vs_addition.compats.computercraft.PeripheralCommon.registerGenericPeripheralCommon
import io.github.xiewuzhiying.vs_addition.fabric.compats.computercraft.FabricPeripheralLookup.peripheralProvider
import io.github.xiewuzhiying.vs_addition.fabric.stuff.FakeRenderer
import net.fabricmc.fabric.api.client.rendering.v1.WorldRenderEvents
import net.fabricmc.fabric.api.event.lifecycle.v1.ServerLifecycleEvents
import net.minecraft.core.BlockPos
import net.minecraft.core.Direction
import net.minecraft.world.level.Level
import net.minecraft.world.level.block.entity.BlockEntity
import net.minecraft.world.level.block.state.BlockState
import org.valkyrienskies.mod.fabric.common.ValkyrienSkiesModFabric

object VSAdditionModFabric {
    fun onInitialize() {
        ValkyrienSkiesModFabric().onInitialize();
        init()
        if (CC_ACTIVE) {
            registerGenericPeripheralCommon()
            VSAdditionModFabricPeripheralLookup.registerFallback()
        }
        ServerLifecycleEvents.SERVER_STARTING.register { server ->
            PlatformUtilsImpl.minecraft = server
        }
    }

    fun onInitializeClient() {
        initClient()
        WorldRenderEvents.BEFORE_DEBUG_RENDER.register(FakeRenderer())
        WorldRenderEvents.LAST.register(FakeRenderer())
    }
}